cli/azd/pkg/project/service_target_aks.go

```diff
@@ -265,6 +265,17 @@ func (t *aksTarget) Deploy(
 		return ServiceDeploymentResult{}, err
 	}
 
+	if len(endpoints) > 0 {
+		// The AKS endpoints contain some additional identifying information
+		// Split on common to pull out the URL as the first segment
+		// The last endpoint in the array will be the most publicly exposed
+		endpointParts := strings.Split(endpoints[len(endpoints)-1], ",")
+		t.env.SetServiceProperty(t.config.Name, "ENDPOINT_URL", endpointParts[0])
+		if err := t.env.Save(); err != nil {
```

Comment:
Do we ever have a use-case when we don't call `env.Save()` ??

I wonder if we should make the `Save()` to be automatically trigger after each update to `env`.   

I am thinking if based on the way we use/require `environment` in azd, there might be a way to avoid introducing the concept of `Reloading`.  I am suspecting that soon, we will see `env.Reload()` being called from everywhere...  just to make sure we are not missing updates... 

Would it make sense, for example, to make `Reload` an implementation detail from `Save()` and keep it unexported?  

Maybe the introduction to `hooks` is enough reason to stop cashing the env in memory? As now we must refresh the cash after any call to a hook...

Reply:
I like the idea of the implicit save on any mutation of the values.  Added more feedback in the comment above.  Too big of a change to take on for this PR but would love to get it in for GA.